### ISSUE TITLE
Grid Path Breakpoint 

### ISSUE THREAD

**Issue by iKarimEssam** [anseki/leader-line#227]
Hello,
is there any way i make the Grid path breakpoint happen before the endElement 100px, not from the middle?

**Comment by anseki** [comment]
Hi @iKarimEssam, thank you for the comment.
Could you explain more?
Is this related to your another issue(https://github.com/anseki/leader-line/issues/226)?
Also, was that issue solved?

**Comment by iKarimEssam** [comment]
yes I have fixed the first issue,
here is a screenshot of what i want to make 
![Screenshot 2021-05-05 04-02-38](https://user-images.githubusercontent.com/83616512/117089968-4ec18f00-ad57-11eb-9d64-9e95b91232f9.png)

**Comment by anseki** [comment]
You can do that by using `startSocketGravity` / `endSocketGravity` options.

**Comment by iKarimEssam** [comment]
i couldn't make it work as i wanted 
can you send me the code for making it appear before the end element 100px ?

please note i don't know the space between elements

**Comment by iKarimEssam** [comment]
![Untitleds](https://user-images.githubusercontent.com/83616512/117092002-56843200-ad5d-11eb-8f29-703f6d8cc3de.png)

**Comment by iKarimEssam** [comment]
![Untitled2](https://user-images.githubusercontent.com/83616512/117092670-40777100-ad5f-11eb-96be-1c16fff29409.png)

**Comment by anseki** [comment]
Do you mean that you can't understand how to use the options?

**Comment by anseki** [comment]
For example: https://jsfiddle.net/yp5zgnt7/

**Comment by iKarimEssam** [comment]
.....

**Comment by anseki** [comment]
?

**Comment by iKarimEssam** [comment]
as you can see here when i changed the  startSocket to top it no longer work 
https://jsfiddle.net/ebgL8290/

**Comment by anseki** [comment]
.....

**Comment by iKarimEssam** [comment]
1- i need to make the line break before the endElement 100px
2- no matter the distance between the two elements it always breaks before the endElement 
![Screenshot 2021-05-05 06-34-12](https://user-images.githubusercontent.com/83616512/117097723-b7673680-ad6c-11eb-879b-93d99b4a241f.png)

**Comment by anseki** [comment]
Sorry, my English is poor.
Did you try this?: https://github.com/anseki/leader-line/issues/227#issuecomment-832392698

**Comment by iKarimEssam** [comment]
can you update this example to look like the image 
https://jsfiddle.net/ebgL8290/
![Screenshot 2021-05-05 06-34-12](https://user-images.githubusercontent.com/83616512/117099283-1169fb00-ad71-11eb-9c25-a52195289b61.png)

**Comment by iKarimEssam** [comment]
the layout in this image, is this something cant be done ?
![Untitled2](https://user-images.githubusercontent.com/83616512/117099837-82f67900-ad72-11eb-993a-87a7aed88905.png)

**Comment by anseki** [comment]
So, did you try this?: https://github.com/anseki/leader-line/issues/227#issuecomment-832392698

**Comment by iKarimEssam** [comment]
this is not what i want !

**Comment by iKarimEssam** [comment]
my request is very simple i want to make the line look like this ..
i don't want the line go down from the center i want it to go down at the end before it reaches the endElement within 100px
![Untitled2](https://user-images.githubusercontent.com/83616512/117100773-e97c9680-ad74-11eb-90a6-4a420dfbf535.png)

**Comment by anseki** [comment]
My question is very simple. did you try the example?
Because you can customize that for your request easily.
Then, I asked the question repeatedly to help you, but you never answer my question. I don't know why...

**Comment by iKarimEssam** [comment]
Dear Abseki,
i told you this is not what i want to do ..

**Comment by iKarimEssam** [comment]
in your example, 
1- if you changed the startSocket to  'top'  you will find the vertical line will go back to the center

**Comment by iKarimEssam** [comment]
i don't want the vertical line to go down from the center i want to go down when it becomes close to the endElement

**Comment by anseki** [comment]
As I said repeatedly, you can customize the example for your request easily.
Therefore I asked the question repeatedly to help you, but you never answer my question. I don't know why...
Anyway, you can edit the path data to make that you want.
Do you mean that you can't edit the path data?
Could you answer my question?

**Comment by iKarimEssam** [comment]
i swear to god that i have looked at your example and it not what i want,


can you send me an example that looks like this image?
i don't want it to be dragged or anything 
all that i want is the following: 
 the line starts from the top of element one then move all the way to the right then go down and connect from the left of element two  same as this image
![Untitled2](https://user-images.githubusercontent.com/83616512/117103470-b89f6000-ad7a-11eb-82c1-0acdabe7c821.png)

**Comment by anseki** [comment]
You answered me for the first time. 
Anyway, as I said repeatedly, you can edit the path data of my example for your request.
Do you mean that you don't know the path data?

**Comment by anseki** [comment]
If you can't edit the path data, you can add another element for the path.
For example: https://jsfiddle.net/v1k7ozns/
Note that this is not best way.

**Comment by anseki** [comment]
Sorry
:smile:

**Comment by poke** [comment]
I found this issue while looking for a solution for a similar problem.

I believe the miscommunication in this issue happened because the behavior of LeaderLine is different depending on the socket direction you choose for the start element. In the simple case, where the start socket is on the right, calculating the distance between the start and the end element is fine to determine the correct `startSocketGravity` in order to produce the desired result.

However, when you choose the start socket `top`, then the `startSocketGravity` no longer controls where the grid intersection happens on the horizontally but only the vertical intersection (above the start element) is impacted:

![startSocketGravity affecting the vertical distance](https://user-images.githubusercontent.com/132240/210794171-421ce3e9-3f9c-4654-a418-c201b61297f6.png)

So the point leading to the end element is now only controlled by the `endSocketGravity` which alone makes it difficult to ensure a specific distance.

---

I have updated the example and replaced the dragging functionality with a static calculation, and added a second example that shows @iKarimEssam’s issue when the start socket is specified to be `top` instead: https://jsfiddle.net/e1jnpb47/ (code also included here)

```javascript
// example 1
const elmStart = document.getElementById('start');
const elmEnd = document.getElementById('end');
new LeaderLine(elmStart, elmEnd, {
  path: 'grid',
  startSocket: 'right',
  endSocket: 'left',
  // calculate start socket gravity to be the exact needed length for the remainder of the distance to be 100
  startSocketGravity: elmEnd.getBoundingClientRect().left - elmStart.getBoundingClientRect().right - 100,
  endSocketGravity: 100,
});

// example 2
const elmStart2 = document.getElementById('start2');
const elmEnd2 = document.getElementById('end2');
new LeaderLine(elmStart2, elmEnd2, {
  path: 'grid',
  startSocket: 'top', // start socket is now top
  endSocket: 'left',
  startSocketGravity: 50, // this number no longer has an effect of where the grid intersection will be
  endSocketGravity: 100,
});
```

I am not sure if the issue in second example is solvable with the current implementation. Luckily, the first example is exactly my use case, so this issue has at least helped me – Thank you! 😊
